### PR TITLE
sort numbers: built-in-sort

### LINKED ISSUE #15: sort-numbers: built-in array methods

### PR DESCRIPTION
**Pull request by rayanejsilva**
Before the reviewer merges this PR to main/master, all of these boxes must be checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [x] The documentation is rendered into the README
  - [x] Docs were generated just before pushing
  - [x] Docs in the README matches the JSDoc comment
  - [x] The README contains at least one use case
- **Unit Tests**
  - [x] Unit tests are present
  - [x] There are at least 8 test cases
  - [x] All `describe`s and `it`s have clear and helpful names
  - [x] All test names actually describe what is being tested
  - [x] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [x] Side-effects are tested, if necessary
  - [x] Tests were run just before pushing
  - [x] All tests pass
  - [x] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [x] Variables have names that help understand the strategy
  - [x] Spacing and formatting help to understand the strategy
  - [x] The code is written as simply and clearly as possible
  - [x] Use short, clear comments to describe the strategy
- **README**
  - [x] There is an explanation of the solution's strategy in the README
  - [x] The explanation is clear and understandable
  - [x] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] The code was formatted just before pushing
  - [x] No errors occurred while formatting the code
- **Linting**
  - [x] There should be no linting errors or warnings
  - [x] Spelling should be correct in all code, comments and docstrings
  - [x] The code was linted just before pushing
- **The solution file**
  - [x] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments outside of the function body (you can use `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about the solution's strategy
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

### CODE REVIEW

File: src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      const received = sortArrayOfNumbers([25, 45, 10, 3, 81]);
+      const expected = [3, 10, 25, 45, 81];
+      expect(received).toEqual(expected);
+    });
+    it('decimal numbers', () => {
+      const received = sortArrayOfNumbers([9.4, 0.01, 4.1, 1.11]);
+      const expected = [0.01, 1.11, 4.1, 9.4];
+      expect(received).toEqual(expected);
+    });
+    it('mixed whole and decimal numbers', () => {
```
Comment (katsmamina) [inline src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js:15]: here, the example only includes decimal numbers, while it's described as mixed whole and decimal

File: src/sort-numbers/built-in-sort-rayane/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Sort Numbers: use a built-in array method
+
+Takes in an array of numbers and returns a new array with the same numbers sorted from smallest to largest.
+
+---
+
+## Strategy
+
+I chose to add a slice method before passing the compare function to the sort method to prevent the original array to be modified.
```
Comment (katsmamina) [inline src/sort-numbers/built-in-sort-rayane/README.md]: as far as I remember, in Strategy we don't need to to mention the name of methods used, it is left for the Implementation part.

File: src/sort-numbers/built-in-sort-rayane/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+---
+
+## Implementation
+
+- .splice() - to keep the original array intact and prevent it from being changed.
```
Comment (katsmamina) [inline src/sort-numbers/built-in-sort-rayane/README.md]: it's .slice() not .splice() in the code itself

File: src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      const received = sortArrayOfNumbers([4.52, 0.2, 9.7]);
```
Comment (katsmamina) [inline src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js]: here the test example only includes decimal numbers while in line 15 it's mentioned that it should include  whole and decimal numbers.

File: src/sort-numbers/built-in-sort-rayane/README.md (hunk 2 above)
Comment (katsmamina) [reply src/sort-numbers/built-in-sort-rayane/README.md]: also, it's not mentioned that all non-finite numbers should be removed from the array (to pass edge cases)

**Review by katsmamina — CHANGES_REQUESTED** [review]
Good job with a few comments to improve

File: src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js (hunk 4 above)
Comment (rayanejsilva) [reply src/sort-numbers/built-in-sort-rayane/sort-numbers-rayane.spec.js]: @katsmamina Thanks for your comments. I pushed my changes. Can you check it again?

**Review by katsmamina — APPROVED** [review]
approved! :)
